server/bootstrap/logs.js

```diff
@@ -0,0 +1,32 @@
+var morgan = require('morgan');
+var winston = require('winston');
+var fs = require('fs');
+
+var logsDirPath = process.env.ROOT + "/server/data/logs";
+
+if (!fs.existsSync(logsDirPath)) {
+  fs.mkdirSync(logsDirPath, "0755");
+}
+
+var getMorganLog = function(subConfig){
+  subConfig.options.stream = fs.createWriteStream(logsDirPath + '/' + subConfig.options.stream, {flags: 'a'});
+  return morgan(subConfig.format, subConfig.options);
+};
+
+var getWinstonLog = function(subConfig){
+  subConfig.file.filename = logsDirPath + "/" + subConfig.file.filename;
+  return new winston.Logger({
+      transports: [
+        new winston.transports.Console(subConfig.console),
+        new winston.transports.File(subConfig.file)
+      ],
+      exitOnError: false
+  });
+};
+
+module.exports.init = function(config){
+  module.exports.access = getMorganLog(config.log.access);
+  module.exports.error = getWinstonLog(config.log.error);
+  module.exports.manual = getWinstonLog(config.log.manual);
```

Comment:
The reason we use two different log libraries:

1. `morgan` is very light-weight. Better for accessing logging, which is done on every request.
2. `winston` is more complex and robust, and with better stack trace handling, making it better for logging severe errors.